cli/src/add_url.rs

```diff
@@ -0,0 +1,18 @@
+// File: add_url.rs
+use db::PgPool;
+use color_eyre::Result;
+
+use crate::check_auth_status;
+
+pub async fn add_url(pool: &PgPool, url: &str, allow_existing: bool) -> Result<()> {
+
+    check_auth_status(pool).await?;
+
+    let result = db::add_url(pool, url, &allow_existing).await;
+    match result {
+        Ok(message) if message.contains("re-adding is allowed") => println!("{message}"),
+        Ok(message) => println!("URL added successfully with ID: {message}"),
+        Err(e) => println!("Error: {e}"),
+    }
+    Ok(())
+}
```

Comment:
this function call the add_url from the DB that is used in the result variable to match certain outputs for the user.

In this section if the user enter a url that is not in the DB however they pass `-- allow existing`. The system will recognize that the URL wasn't present and automatically add it. 

The output would display `URL added successfully with ID: <url>`

We could go into more depth and output a different output such as 
`Url  wasn't present in the DB, adding it now..... You can now use -- allow-existing `

Reply:
I like the general output you have now! I don't think it needs to be more in depth really, but I like the suggestion!